### PR TITLE
Upgrade code analysis to 1.0.0rc1

### PR DESCRIPTION
**Pull request by pdelvo**
Do not merge.

Currently all SA1400 tests that require a syntax node action of type EventFieldDeclaration or FieldDeclaration being called are failing.

### CODE REVIEW

**Comment by pdelvo** [comment]
Im pretty sure that it is a bug in roslyn. I opened an issue for that https://github.com/dotnet/roslyn/issues/565.

**Comment by sharwell** [comment]
So yes it's a bug in Roslyn. How should we handle it in the meantime?

**Comment by pdelvo** [comment]
I think we should merge this. The bug does not crash anything and it does not produce unnecessary diagnostics. Currently this project does not work in ctp 6 at all so having most of the diagnostics work is an improvment. The bug is fixed so the unit tests and the diagnostics should work as expected in 1.0.0 rc2. 

Should I add an ignore  attribute (and a WorkItem attribute) to the test methods?

**Comment by sharwell** [comment]
No need to create a new work item attribute at this time. I imagine we will eventually switch to xUnit.net  and create one at that time.

Yes, we'll need to ignore these:

```csharp
[Ignore("https://github.com/dotnet/roslyn/issues/565")]
```

**Comment by pdelvo** [comment]
I disabled the tests and created a new issue so that i could link to it. The visual studio test framework does have an attribute that we could use (WorkItemAttribute), but it only takes an integer as parameter so I would not be able to put a link in there.

**Comment by pdelvo** [comment]
I rebased the branch on master because the merged built failed because there was work added on master that used CSharpKind() instead of Kind()

File: StyleCop.Analyzers/StyleCop.Analyzers.Test/MaintainabilityRules/SA1400UnitTests.cs [hunk 1]
```diff
@@ -487,37 +488,37 @@ public class SA1400UnitTests : CodeFixVerifier
 
         #region EventFieldDeclarationSyntax
 
-        [TestMethod]
+        [TestMethod, Ignore, OpenIssue("https://github.com/DotNetAnalyzers/StyleCopAnalyzers/issues/496")]
```
Comment (sharwell) [inline StyleCop.Analyzers/StyleCop.Analyzers.Test/MaintainabilityRules/SA1400UnitTests.cs]: :bulb: I typically prefer these one per line.

File: StyleCop.Analyzers/StyleCop.Analyzers/OrderingRules/SA1212PropertyAccessorsMustFollowOrder.cs [hunk 2]
```diff
@@ -91,8 +91,8 @@ private static void AnalyzeProperty(SyntaxNodeAnalysisContext context, BasePrope
                 return;
             }
 
-            if (accessors[0].CSharpKind() == SyntaxKind.SetAccessorDeclaration &&
-                accessors[1].CSharpKind() == SyntaxKind.GetAccessorDeclaration)
+            if (accessors[0].Kind() == SyntaxKind.SetAccessorDeclaration &&
```
Comment (sharwell) [inline StyleCop.Analyzers/StyleCop.Analyzers/OrderingRules/SA1212PropertyAccessorsMustFollowOrder.cs]: :bulb: This is a good place for `IsKind`.

**Comment by sharwell** [comment]
It looks good :+1: 

The last items I can easily fix during a merge later today.

**Pull request merged**